src/core/Akka.Cluster.Tests/SplitBrainResolverConfigSpec.cs

```diff
@@ -0,0 +1,122 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SplitBrainResolverConfigSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using Akka.Actor;
+using Akka.Configuration;
+using FluentAssertions;
+using Xunit;
+
+namespace Akka.Cluster.Tests
+{
+    public class SplitBrainResolverConfigSpec
+    {
+        [Fact]
+        public void StaticQuorum_must_parse_settings_from_config()
+        {
+            var config = ConfigurationFactory.ParseString(@"
+                quorum-size = 5
+                role = test-role");
+            var strategy = new StaticQuorum(config);
+
+            strategy.QuorumSize.Should().Be(5);
+            strategy.Role.Should().Be("test-role");
+        }
+
+        [Fact]
+        public void KeepMajority_must_parse_settings_from_config()
+        {
+            var config = ConfigurationFactory.ParseString(@"
+                role = test-role");
+            var strategy = new KeepMajority(config);
+            
+            strategy.Role.Should().Be("test-role");
+        }
+
+        [Fact]
+        public void KeepOldest_must_parse_settings_from_config()
+        {
+            var config = ConfigurationFactory.ParseString(@"
+                down-if-alone = true
+                role = test-role");
+            var strategy = new KeepOldest(config);
+
+            strategy.Role.Should().Be("test-role");
+            strategy.DownIfAlone.Should().Be(true);
+        }
+
+        [Fact]
+        public void KeepReferee_must_parse_settings_from_config()
+        {
+            var config = ConfigurationFactory.ParseString(@"
+                address = ""akka.tcp://system@localhost:5000/""
+                down-all-if-less-than-nodes = 3");
+            var strategy = new KeepReferee(config);
+
+            strategy.Role.Should().BeNull();
+            strategy.DownAllIfLessThanNodes.Should().Be(3);
+            strategy.Address.Should().Be(Address.Parse("akka.tcp://system@localhost:5000/"));
+        }
+
+        [Fact]
+        public void SplitBrainResolver_should_be_picked_up_if_set_as_downing_provider()
+        {
+            var config = ConfigurationFactory.ParseString(@"
+                akka {
+                    actor.provider = cluster
+                    cluster {
+                        down-removal-margin = 10s
+                        downing-provider-class = ""Akka.Cluster.SplitBrainResolver, Akka.Cluster""
```

Comment:
If someone specifies a `downing-provider-class` and an active `split-brain-resolver` strategy, which one wins?

Reply:
A downing provider is an actual actor class that host all necessary logic i.e. gathering the information about the last known reachable and unreachable members, downing them, waiting for cluster to become stable etc.

A split brain resolver strategy is an actual decider that based on its own configuration and last known state of the cluster must make a decision, which nodes should be downed.

For a split brain to work you must provide both of them: `SplitBrainResolver` as a `downing-provider-class` and a particular implementation of `split-brain-resolver.active-strategy`.